### PR TITLE
Rust improvements

### PR DESCRIPTION
**Pull request by lights0123**
Fixes some, but not all issues mentioned in #56. I made my changes in small commits, so you can see what I did step-by-step.

### CODE REVIEW

**Comment by lights0123** [comment]
Note that this is a breaking change, so when you publish to crates.io, you must change the version to 0.2.0.

File: source/ports/rs_port/src/lib.rs [hunk 1]
```diff
@@ -18,288 +18,303 @@
...
+                Any::Int(x) => metacall_value_create_int(*x),
+                Any::Long(x) => metacall_value_create_long(*x),
+                Any::Float(x) => metacall_value_create_float(*x),
+                Any::Double(x) => metacall_value_create_double(*x),
+                Any::Bool(x) => metacall_value_create_bool(*x as c_int),
+                Any::Char(x) => metacall_value_create_char(*x as c_char),
+                Any::Str(x) => {
+                    let st = CString::new(x.as_str()).expect("can't convert to c str");
+                    metacall_value_create_string(st.as_ptr(), x.len())
+                }
```
Comment (lights0123) [inline source/ports/rs_port/src/lib.rs:176]: The documentation isn't clear on whether `metacall_value_to_string` requires a null-terminated string, as the size is passed along with it. If it isn't required, then

```suggestion
                Any::Str(x) => metacall_value_create_string(x.as_ptr() as *const _, x.len()),
```

Comment (viferga) [reply source/ports/rs_port/src/lib.rs:176]: I have added more doc about that: https://github.com/metacall/core/commit/51b85c876df753f1b89738cc253fefb9bacd9984

I also realized that metacall_value_from_string is very dangerous if not used properly, it truncates the size if the size is smaller (without leaving null terminated string), or it leads a gap in the end if the size is bigger (but in this case it is null terminated). I don't know what implications can have this, but when using the API to implement the Ports, I try to copy values when possible and I threat it as what it is, a C API.

**Comment by viferga** [comment]
> Note that this is a breaking change, so when you publish to crates.io, you must change the version to 0.2.0.

Published with version 0.2.0: https://crates.io/crates/metacall

**Pull request merged**